docs/root/configuration/http/http_filters/local_rate_limit_filter.rst

```diff
@@ -103,6 +103,83 @@ The route specific configuration:
 Note that if this filter is configured as globally disabled and there are no virtual host or route level
 token buckets, no rate limiting will be applied.
 
+Using Descriptors to rate limit on
+----------------------------------
+
+Descriptors can be used to override local rate limiting based on presence of certain descriptors/route actions.
+A routes :ref:`rate limit action <envoy_v3_api_msg_config.route.v3.RateLimit>` is used to match up a
+:ref:`descriptor <envoy_v3_api_msg_extensions.common.ratelimit.v3.RateLimitDescriptor>` in the filter config descriptor list.
+The descriptor's rate limit override value is used to specify the rate limit units to use if this descriptor is a match
+otherwise token bucket default config is used.
+
+Example filter configuration using descriptors is as follows:
+
+.. code-block:: yaml
+
+  route_config:
+    name: local_route
+    virtual_hosts:
+    - name: local_service
+      domains: ["*"]
+      routes:
+      - match: { prefix: "/foo" }
+        route: { cluster: service_protected_by_rate_limit }
+        typed_per_filter_config:
+          envoy.filters.http.local_ratelimit:
+            "@type": type.googleapis.com/envoy.extensions.filters.http.local_ratelimit.v3.LocalRateLimit
+            stat_prefix: test
+            token_bucket:
```

Comment:
I wonder if it would be better to treat route name as another descriptor?

Reply:
But that should be for like global rate limiting right? Here filter is per route

Reply:
I was thinking of host/filter level descriptor that can reference the route name. It could replace per-route config.

Reply:
per-route config is similar to what http global rate limiter has. Thus, this would be changing that.

@mattklein123 : what do you think?

Reply:
Sorry what do I think about what? I think what @kyessenov is suggesting is orthogonal to this change? We can add other descriptor types but it's unrelated to what we are doing here? Or are you saying you don't want to do this anymore?

Reply:
@mattklein123 : I think there is definitely value in what we are doing here.
@kyessenov : we can add what you are suggesting in a different PR?

Reply:
Yes, adding route name in a different PR is fine with me.